--- origsrc/xclock-1.0.5/Clock.c	2010-09-03 18:04:42.000000000 -0500
+++ src/xclock-1.0.5/Clock.c	2011-08-23 16:36:32.543486400 -0500
@@ -2214,7 +2214,7 @@ clock_to_utf8(const char *str, int in_le
     char *buf;
     size_t buf_size;
     size_t ileft, oleft;
-    const char *inptr;
+    ICONV_CONST char *inptr;
     char *outptr;
     size_t ret;
     const char *code_set = nl_langinfo(CODESET);
--- origsrc/xclock-1.0.5/configure.ac	2010-09-23 20:33:53.000000000 -0500
+++ src/xclock-1.0.5/configure.ac	2011-08-23 16:34:04.563282200 -0500
@@ -43,11 +43,11 @@ AC_PROG_INSTALL
 
 AC_CHECK_FUNCS([strlcpy getpid])
 
-AC_SEARCH_LIBS([iconv], [iconv],
-	[AC_DEFINE(HAVE_ICONV, 1, [Define to 1 if iconv() is available.])])
+AM_ICONV
 
 dnl Required dependencies
 PKG_CHECK_MODULES(XCLOCK, xaw7 xmu x11)
+XCLOCK_LIBS="$XCLOCK_LIBS $LIBICONV"
 
 dnl Optional dependencies
 AC_ARG_WITH(xft, AC_HELP_STRING([--with-xft],[Use Xft2 and Xrender for rendering (Default is YES)]),use_xft="$withval",use_xft="try")
